grpc 1.63.0 fails to build with gcc 14

Upstream-issue: https://github.com/grpc/grpc/issues/35945
Upstream-issue: https://github.com/grpc/grpc/issues/37731
diff --git a/Gemfile b/Gemfile
index bc62c3c4bfc2..19c598e4fa7f 100644
--- a/Gemfile
+++ b/Gemfile
@@ -648,9 +648,7 @@ gem 'gitaly', '~> 18.1.0.pre.rc1', feature_category: :gitaly
 # KAS GRPC protocol definitions
 gem 'gitlab-kas-grpc', '~> 17.11.0', feature_category: :deployment_management
 
-# Lock until 1.74.0 is available
-# https://gitlab.com/gitlab-com/gl-infra/production/-/issues/20067
-gem 'grpc', '= 1.63.0', feature_category: :shared
+gem 'grpc', '~> 1.71', feature_category: :shared
 
 gem 'google-protobuf', '~> 3.25', '>= 3.25.3', feature_category: :shared
 
diff --git a/Gemfile.lock b/Gemfile.lock
index 523b70bcb251..63e1ef4e5cfa 100644
--- a/Gemfile.lock
+++ b/Gemfile.lock
@@ -48,7 +48,7 @@ PATH
       google-cloud-storage_transfer (~> 1.2.0)
       google-protobuf (~> 3.25, >= 3.25.3)
       googleauth (~> 1.8.1)
-      grpc (= 1.63.0)
+      grpc (~> 1.71)
       json (~> 2.7)
       jwt (~> 2.5)
       logger (~> 1.5)
@@ -966,7 +966,7 @@ GEM
       graphql (~> 2.0)
       html-pipeline (~> 2.14, >= 2.14.3)
       sass-embedded (~> 1.58)
-    grpc (1.63.0)
+    grpc (1.71.0)
       google-protobuf (~> 3.25)
       googleapis-common-protos-types (~> 1.0)
     grpc-google-iam-v1 (1.5.0)
@@ -2214,7 +2214,7 @@ DEPENDENCIES
   graphlyte (~> 1.0.0)
   graphql (= 2.4.13)
   graphql-docs (~> 5.0.0)
-  grpc (= 1.63.0)
+  grpc (= 1.71.0)
   gssapi (~> 1.3.1)
   guard-rspec
   haml_lint (~> 0.58)
diff --git a/gems/gitlab-backup-cli/Gemfile.lock b/gems/gitlab-backup-cli/Gemfile.lock
index 690dc5b1d255..a8ee40162be6 100644
--- a/gems/gitlab-backup-cli/Gemfile.lock
+++ b/gems/gitlab-backup-cli/Gemfile.lock
@@ -19,7 +19,7 @@ PATH
       google-cloud-storage_transfer (~> 1.2.0)
       google-protobuf (~> 3.25, >= 3.25.3)
       googleauth (~> 1.8.1)
-      grpc (= 1.63.0)
+      grpc (= 1.71.0)
       json (~> 2.7)
       jwt (~> 2.5)
       logger (~> 1.5)
@@ -106,7 +106,7 @@ GEM
       multi_json (~> 1.11)
       os (>= 0.9, < 2.0)
       signet (>= 0.16, < 2.a)
-    grpc (1.63.0)
+    grpc (1.71.0)
       google-protobuf (~> 3.25)
       googleapis-common-protos-types (~> 1.0)
     grpc (1.63.0-aarch64-linux)
diff --git a/gems/gitlab-backup-cli/gitlab-backup-cli.gemspec b/gems/gitlab-backup-cli/gitlab-backup-cli.gemspec
index bf4311713f6a..c753c939ff03 100644
--- a/gems/gitlab-backup-cli/gitlab-backup-cli.gemspec
+++ b/gems/gitlab-backup-cli/gitlab-backup-cli.gemspec
@@ -40,7 +40,7 @@ Gem::Specification.new do |spec|
   spec.add_dependency "concurrent-ruby", "~> 1.1"
   spec.add_dependency "faraday", "~> 2"
   spec.add_dependency "google-protobuf", "~> 3.25", ">= 3.25.3"
-  spec.add_dependency "grpc", "= 1.63.0"
+  spec.add_dependency "grpc", "= 1.71.0"
   spec.add_dependency "json", "~> 2.7"
   spec.add_dependency "jwt", "~> 2.5"
   spec.add_dependency "logger", "~> 1.5"
